### PR TITLE
Uplift pull request 7245 (Brave Today p3a) to 1.18.x

### PR DESCRIPTION
**Pull request by petemill**
Uplift https://github.com/brave/brave-core/pull/7245 to 1.18.x

### CODE REVIEW

**Comment by petemill** [comment]
All CI passed except Windows which had an instance error. There's nothing platform specific so we're ok IMO

**Review by iefremov — APPROVED** [review]

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.18.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/7245#issuecomment-741123918. However, `Brave.Today.WeeklyMaxCardViewsCount` & `Brave.Today.WeeklyMaxCardVisitsCount` weren't working and will be addressed via  https://github.com/brave/brave-browser/issues/13096 & https://github.com/brave/brave-browser/issues/13097.

**Pull request merged**
